metatensor-core/src/labels.rs

```diff
@@ -326,7 +208,98 @@ impl std::fmt::Debug for Labels {
     }
 }
 
+fn init_positions(values: &[LabelValue], size: usize) -> HashMap<LabelsEntry, usize, AHashHasher> {
+    assert!(values.len() % size == 0);
+
+    let mut positions = HashMap::new();
+    for (i, entry) in values.chunks_exact(size).enumerate() {
+        // entries should be unique!
+        positions.insert_unique_unchecked(entry.into(), i);
+    }
+    return positions;
+}
+
 impl Labels {
+    /// Create new Labels with the given names and values.
+    ///
+    /// The values are given as a flatten, row-major array, and we will check
+    /// that rows are unique in the array.
+    pub fn new(names: &[&str], values: Vec<impl Into<LabelValue>>) -> Result<Labels, Error> {
+        let values = values.into_iter().map(Into::into).collect();
+        return Labels::new_impl(names, values, true);
+    }
+
+    /// Create new labels with the given names and values.
+    ///
+    /// This is identical to [`Labels::new`] except that the rows are not
+    /// checked for uniqueness, but instead the caller must ensure that rows are
+    /// unique.
+    pub unsafe fn new_unchecked_uniqueness(names: &[&str], values: Vec<impl Into<LabelValue>>) -> Result<Labels, Error> {
```

Comment:
I agree that, in the future, it would be interesting to make it part of the API(s) (not only for users that know their labels are unique, but also for those that have a faster way to check for uniqueness, e.g. on an accelerator)